lib/circuits_sim/device/sht4x.ex

```diff
@@ -0,0 +1,96 @@
+defmodule CircuitsSim.Device.SHT4X do
+  @moduledoc """
+  Sensirion SHT4x sensors
+
+  Typically found at 0x44
+  See the [datasheet](https://cdn-learn.adafruit.com/assets/assets/000/099/223/original/Sensirion_Humidity_Sensors_SHT4x_Datasheet.pdf) for details.
+
+  Call `set_humidity_rh/3` and `set_temperature_c/3` to change the state of the sensor.
+  """
+  alias CircuitsSim.I2C.I2CDevice
+  alias CircuitsSim.I2C.I2CServer
+
+  defstruct current: nil, humidity_rh: 50.0, temperature_c: 20.0
+  @type t() :: %__MODULE__{current: atom(), humidity_rh: float(), temperature_c: float()}
+
+  @spec child_spec(keyword()) :: Supervisor.child_spec()
+  def child_spec(args) do
+    device = __MODULE__.new()
+    I2CServer.child_spec_helper(device, args)
+  end
+
+  @spec new() :: t()
+  def new() do
+    %__MODULE__{}
+  end
+
+  @spec set_humidity_rh(String.t(), Circuits.I2C.address(), float()) :: :ok
+  def set_humidity_rh(bus_name, address, humidity_rh) when is_number(humidity_rh) do
+    I2CServer.send_message(bus_name, address, {:set_humidity_rh, humidity_rh})
+  end
+
+  @spec set_temperature_c(String.t(), Circuits.I2C.address(), float()) :: :ok
+  def set_temperature_c(bus_name, address, temperature_c) when is_number(temperature_c) do
+    I2CServer.send_message(bus_name, address, {:set_temperature_c, temperature_c})
+  end
+
+  ## protocol implementation
+
+  defimpl I2CDevice do
+    @impl I2CDevice
+    def read(%{current: :serial_number} = state, 6) do
+      {<<15, 186, 124, 249, 143, 14>>, %{state | current: nil}}
+    end
```

Comment:
How about something like this so that the correct number of bytes are returned:

```suggestion
    def read(%{current: :serial_number} = state, count) do
      result = <<15, 186, 124, 249, 143, 14>> |> trim_pad(count)
      {result, %{state | current: nil}}
    end
    
    defp trim_pad(x, count) when byte_size(x) >= count, do: :binary.part(x, 0, count)
    defp trim_pad(x, count), do: x <> :binary.copy(<<0>>, count - byte_size(x))
```